mesh/mesh.go

```diff
@@ -219,15 +219,12 @@ func (msh *Mesh) setProcessedLayer(layerID types.LayerID) error {
 // the block in consensus, and reverts state before that layer
 // if such layer is found.
 func (msh *Mesh) ensureStateConsistent(ctx context.Context, results []result.Layer) error {
-	var (
-		changed types.LayerID
-		inState = msh.LatestLayerInState()
```

Comment:
the question is... how did inState become incorrect?

Reply:
i found two ways:
- if loop that updates inState layer is terminated with error, we would update it to some outdated layer
- or if loop terminated earlier, for example we applied 10, 11, 12 (12 because of hare), later hare falls out of hdist window and not yet verified by tortoise because of partition (opinion on that layer is abstain)